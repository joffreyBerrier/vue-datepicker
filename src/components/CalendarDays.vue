<script lang="ts">
export default {
  name: "CalendarDays",
};
</script>

<script setup lang="ts">
import { computed, ref, inject, type ComputedRef } from "vue";
import type { Ref } from "vue";

import type { HeaderDay } from "../types";

const props = defineProps({
  locale: {
    type: String,
    required: true,
  },
});

const t = inject("t", (key: string) => ({}));

const listOfDays: Ref<HeaderDay[]> = ref([
  { key: 1, name: "monday" },
  { key: 2, name: "tuesday" },
  { key: 3, name: "wednesday" },
  { key: 4, name: "thursday" },
  { key: 5, name: "friday" },
  { key: 6, name: "saturday" },
  { key: 0, name: "sunday" },
]);

const days: ComputedRef<HeaderDay[]> = computed(() => {
  const copyListOfDays = JSON.parse(
    JSON.stringify(listOfDays.value),
  ) as HeaderDay[];

  if (props.locale === "en")
    return copyListOfDays.sort((a, b) => a.key - b.key);

  return listOfDays.value;
});
</script>

<template>
  <ul class="calendar_wrapper_content-header-days">
    <li v-for="day in days" :key="day.key" class="calendar-days">
      {{ t(`days.${day.name}`) }}.
    </li>
  </ul>
</template>

<style>
.vue-calendar .calendar_wrapper_content-header-days {
  @apply grid grid-cols-7 pt-5 pb-1.5;
}
.vue-calendar .calendar-days {
  @apply text-[14px] text-center;
  color: var(--calendar-header-days-color);
}
</style>
