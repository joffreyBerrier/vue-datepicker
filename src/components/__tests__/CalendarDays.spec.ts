import { afterEach, beforeEach, describe, expect, it } from "vitest";

import { mount } from "@vue/test-utils";
import CalendarDays from "../CalendarDays.vue";

let wrapper: any;

beforeEach(() => {
  wrapper = mount(CalendarDays, {
    global: {
      provide: {
        t: (key: string): string => {
          const translation: any = {
            days: {
              monday: "Mo",
              tuesday: "Tu",
              wednesday: "We",
              thursday: "Th",
              friday: "Fr",
              saturday: "Sa",
              sunday: "Su",
            },
          };

          if (key.includes(".")) {
            const a = key.split(".");

            const translationValue = translation[a[0]][a[1]];
            return translationValue;
          } else {
            return translation[key];
          }
        },
      },
    },
  });
});

afterEach(() => {
  wrapper.unmount();
});

describe("CalendarDays", () => {
  it("is a Vue instance", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it("renders CalendarDays properly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
