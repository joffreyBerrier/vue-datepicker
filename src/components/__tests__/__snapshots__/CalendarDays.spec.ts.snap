// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CalendarDays > renders CalendarDays properly 1`] = `
"<ul class="calendar_wrapper_content-header-days">
  <li class="calendar-days">Mo. </li>
  <li class="calendar-days">Tu. </li>
  <li class="calendar-days">We. </li>
  <li class="calendar-days">Th. </li>
  <li class="calendar-days">Fr. </li>
  <li class="calendar-days">Sa. </li>
  <li class="calendar-days">Su. </li>
</ul>"
`;
